blockbuster/abci/abci.go

```diff
@@ -93,35 +109,36 @@ func ChainPrepareLanes(chain ...blockbuster.Lane) blockbuster.PrepareLanesHandle
 		chain = append(chain, terminator.Terminator{})
 	}
 
-	return func(ctx sdk.Context, partialProposal *blockbuster.Proposal) (finalProposal *blockbuster.Proposal) {
+	return func(ctx sdk.Context, partialProposal blockbuster.BlockProposal) (finalProposal blockbuster.BlockProposal, err error) {
 		lane := chain[0]
 		lane.Logger().Info("preparing lane", "lane", lane.Name())
 
 		// Cache the context in the case where any of the lanes fail to prepare the proposal.
 		cacheCtx, write := ctx.CacheContext()
 
 		defer func() {
-			if err := recover(); err != nil {
-				lane.Logger().Error("failed to prepare lane", "lane", lane.Name(), "err", err)
+			if rec := recover(); rec != nil || err != nil {
+				lane.Logger().Error("failed to prepare lane", "lane", lane.Name(), "err", err, "recover_error", rec)
```

Comment:
Could logging the errors lead to any non-determinism? I was talking with Jeremy about IBC-go and he told me that errors thrown internally are ignored because the logs can lead to determinism. I don't have full context into that but wanted to make sure that we can actually log the error strings here.

Reply:
_logging_ cannot lead to non-determinism. Maybe he meant `TxResult.Log`?

Reply:
the huckleberry security patch specifically, idk if its only IBC specific.

Reply:
Yea, it was the result of a cosmwasm tx result log.

Reply:
Yeah Log ≠ Logging :p